File: backend/src/main/java/com/meterease/backend/entity/MeterReading.java
package com.meterease.backend.entity;

import com.meterease.backend.type.UtilityType;
import jakarta.persistence.*;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import lombok.*;

import java.time.LocalDateTime;


@Entity
@Table(name = "meter_readings")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MeterReading {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer meterReadingId;


    @ManyToOne
    @JoinColumn(name = "room_id", nullable = false)
    private Room room;


    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private UtilityType utilityType;


    @Column(nullable = false)
    @Min(0)
    @Max(99999)
    private Integer meterReadingValue;


    private String meterImage;


    @Column(nullable = false)
    private LocalDateTime recordedDateTime;

}

File: frontend/src/types/meterReading.ts
export type UtilityType = 'ELECTRICITY' | 'WATER'

export interface MeterReading {
  meterReadingId: number
  roomId: number
  utilityType: UtilityType
  meterReadingValue: number
  meterImage: string | null
  recordedDateTime: string
}

export interface MeterReadingRequest {
  roomId: number
  utilityType: UtilityType
  meterReadingValue: number
  meterImage?: string | null
}

export interface RoomReadingEntry {
  roomId: number
  roomName: string

  previousElectricity: number | null
  currentElectricity: number | null

  previousWater: number | null
  currentWater: number | null
}